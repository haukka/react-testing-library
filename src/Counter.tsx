import React, { useState } from 'react';

const Counter = () => {
    const [count, setCount] = useState(0);

    const increase = () => {
        setCount(count + 1);
    }

    const decrease = () => {
        setTimeout(() => {
            setCount(count - 1);
        }, 250);
    }

    return (
        <div>
            <button onClick={increase}>Up</button>
            <button onClick={decrease}>Down</button>
            <span data-testid="count">{count}</span>
        </div>
    );
};

export default Counter;