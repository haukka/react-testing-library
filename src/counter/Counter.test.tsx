import React from "react";
import { screen, render, cleanup, fireEvent, waitFor, getByText } from "@testing-library/react";
import Counter from "./Counter";

afterEach(cleanup);

describe('testing Counter Component', () => {
    it('test render', () => {
        const { asFragment } = render(<Counter />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('test the presence of the counter', () => {
        render(<Counter />);
        expect(screen.getByTestId('count')).toHaveTextContent('0');
    });   
    
    it('test the click event for the increment part', () => {
        render(<Counter />);
        fireEvent.click(screen.getByText('Increment'));
        expect(screen.getByTestId('count')).toHaveTextContent('1');
    });

    it('test the click event for the decrement part', async () => {
        render(<Counter />);
        fireEvent.click(screen.getByText('Decrement'));
        const count = await waitFor(() => screen.findByText('-1'));
        expect(count).toHaveTextContent('-1');
    });
});