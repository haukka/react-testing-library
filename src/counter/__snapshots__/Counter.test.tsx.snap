// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Counter Component test render 1`] = `
<DocumentFragment>
  <div>
    <button>
      Increment
    </button>
    <button>
      Decrement
    </button>
    <span
      data-testid="count"
    >
      0
    </span>
  </div>
</DocumentFragment>
`;
